keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -0,0 +1,423 @@
+  
+1. **Performance impact**
+    
+  We will take care to carefully consider the computational overhead associated with the statusz endpoint to avoid negatively impacting the server's overall performance.
+
+2. **Essential data only**
+    
+  We will prioritize inclusion of only essential diagnostic and monitoring data that aligns with the intended purpose of the z-page.
+
+3. **Premature Dependency on Unstable Format**
+
+  The alpha release will explicitly support plain text format only, making it clear that the output is not intended for parsing or automated monitoring. The feature will be secured behind a feature gate that is disabled by default, ensuring users opt-in consciously. Also, the endpoint will be implemented with versioned URLs to facilitate future transitions to structured formats without breaking existing integrations, once the usage patterns and requirements are better understood
+
+## Design Details
+
+### Data Format and versioning
+
+Initially, the statusz page will exclusively support a plain text format for responses. We will implement these endpoints using versioned URLs, like /statusz/v1, to ensure future compatibility. By using versioned URLs, like /statusz/v2, we can evolve the endpoint and introduce changes to the response schema without impacting clients relying on previous versions, ensuring backward compatibility.
+
+### Authz and authn
+
+Access to the endpoint will be limited to members of the `system:monitoring group` ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), which is how paths like /healthz, /livez, /readyz are restricted today. 
+
+### Endpoint Response Format
+
+#### Data format : text
+
+#### Request
+* Method: **GET** 
+* Endpoint: **/statusz/v1**
+* Header: `Content-Type: text/plain`
+* Body: empty
+
+#### Response fields
+* **Start Time**: The exact date and time when the component process was initiated
+* **Uptime**: The duration for which the component has been running continuously
+* **Go Version**: The specific version of the Go programming language used to build the component's binary
+* **Binary Version**: The version of the component's binary, helpful for compatibility checks and troubleshooting
+* **Emulation Version**: The Kubernetes API version which this component is emulating as
+* **Minimum Compatibility Version**: The minimum Kubernetes API version with which the component is designed to work seamlessly
+* **Useful Links**: Relative URLs to other essential endpoints, such as /healthz, /livez, /readyz, and /metrics, for deeper insights into the component's health, readiness, and performance metrics
+
+#### Sample response
+
+```
+Started: Fri Sep  6 06:19:51 UTC 2024
```

Comment:
Should we include PID as well? Or "started" is enough?

Reply:
Is knowing the PID going to be helpful for specific debugging purposes? Also, is it safe to expose this info actually (thinking about signal injection risks) ?

Reply:
I was mostly thinking of a way to "label" information so we can quickly tell if the information is the same or needs to be re-read.

Reply:
Seems like 'started' / 'uptime' should suffice for that then?

Reply:
yes, PID is easier for me to compare. But I can be ok with just Started

Reply:
Most folks run core components with `hostPID: false` AFAIK.  which PID is it going to show though? AFAIK from within the container its almost always going to be 1? The PID on the host would be more interesting but we won't know that from within the container.

Reply:
yeah, I was thinking about kubelet =)